lib/router.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:challenge/navs/thread_main_nav.dart';
+import 'package:challenge/screens/thread_activity_screen.dart';
+import 'package:challenge/screens/thread_home_screen.dart';
+import 'package:challenge/screens/thread_privacy_screen.dart';
+import 'package:challenge/screens/thread_profile_screen.dart';
+import 'package:challenge/screens/thread_search_screen.dart';
+import 'package:challenge/screens/thread_settings_screen.dart';
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:go_router/go_router.dart';
+
+enum MainNavTab {
+  home(
+    icon: FontAwesomeIcons.house,
+    screen: ThreadHomeScreen(),
+  ),
+  search(
+    icon: FontAwesomeIcons.magnifyingGlass,
+    screen: ThreadSearchScreen(),
+  ),
+  post(
+    icon: FontAwesomeIcons.penToSquare,
+  ),
+  activity(
+    icon: FontAwesomeIcons.heart,
+    screen: ThreadActivityScreen(),
+  ),
+  profile(
+    icon: FontAwesomeIcons.person,
+    screen: ThreadProfileScreen(),
+  );
+
+  final Widget? screen;
+  final IconData icon;
+
+  const MainNavTab({this.screen, required this.icon});
+}
+
+class Routes {
+  // route 관리 이렇게 하는게 일반적인지, subRoute는 / 관리 어떻게 하는지 궁금합니다
+  static const home = "/";
+  static const mainNav = "/:tab";
+  static const settings = "/settings";
+  static const privacy = "privacy";
+}
+
+final router = GoRouter(
+  routes: [
+    GoRoute(
+      path: Routes.home,
+      builder: (context, state) => const ThreadMainNav(),
+    ),
+    GoRoute(
+      path: Routes.settings,
```

Comment:
현재 `/` 와 `/home` 모두 같은 화면을 바라보고 있어서 중복되는 route 를 두는 것 보다는 아래와 같이 설정하는 것이 더 좋아보입니다.

```suggestion
  initialLocation: "/${MainNavTab.home.name}",
  routes: [
    GoRoute(
        path: Routes.settings,
```